src/cloud-functions.ts

```diff
@@ -525,5 +526,20 @@ export function optionsToTrigger(options: DeploymentOptions) {
     trigger.vpcConnectorEgressSettings = options.vpcConnectorEgressSettings;
   }
 
+  if (options.serviceAccount) {
+    if (options.serviceAccount === 'default') {
+      // Do nothing, since this is equivalent to not setting serviceAccount.
+    } else if (options.serviceAccount.includes('@')) {
```

Comment:
The approved API for this was actually to explicitly put a trailing `@` to indicate that it is a named service account for the current project. So the `else` should be moved up to an `else if` that comes before this something like:

```ts
    else if (options.serviceAccount.endsWith('@')) {
      if (!process.env.GCLOUD_PROJECT) {
        throw new Error(
          `Unable to determine email for service account '${options.serviceAccount}' because process.env.GCLOUD_PROJECT is not set.`
        );
      }
      trigger.serviceAccountEmail = `${options.serviceAccount}${process.env.GCLOUD_PROJECT}.iam.gserviceaccount.com`;
    }
```